### PR TITLE
Updates to Node.js tutorial and hello, world demo

### PR DESCRIPTION
**Pull request by Ryanmtate**
Updates to the `Hello, World!` example for node client.

### CODE REVIEW

File: content/tutorials/hello-world-node.md [hunk 1]
```diff
@@ -35,320 +36,459 @@ Node.js installation varies depending on your operating system.
...
-The **nj-cli** is a tool that generates native Node interfaces for Rust APIs. Use  **cargo** to install nj-cli tool:
+The Fluvio Node.js client provides TypeScript definitions. Let's setup our demo application to use TypeScript.
 
 ```bash
-$ cargo install nj-cli
-Updating crates.io index
-...
-Installing /Users/user/.cargo/bin/nj-cli
-Installed package `nj-cli v0.1.2` (executable `nj-cli`)    
+# Install TypeScript locally to the project as a development dependency;
```
Comment (nicholastmosher) [inline content/tutorials/hello-world-node.md:65]: I think comments look too dark in the bash code blocks. Can we move the comments to just plain text?

image

File: content/tutorials/hello-world-node.md [hunk 2]
```diff
@@ -35,320 +36,459 @@ Node.js installation varies depending on your operating system.
...
 ```
 
+{{< idea >}}
 
-### Add Fluvio Client library
-
-The client library exports Fluvio data streaming APIs to the Node App.
-
-Use npm install to add <a href="https://www.npmjs.com/package/@fluvio/client" target="_blank">@fluvio/client</a> to the project:
+Your directory should now contain the following:
```
Comment (nicholastmosher) [inline content/tutorials/hello-world-node.md:122]: following files

File: content/tutorials/hello-world-node.md [hunk 3]
```diff
@@ -35,320 +36,459 @@ Node.js installation varies depending on your operating system.
...
+    "@types/node": "^14.11.2",
+    "typescript": "^4.0.3"
+  },
+  "dependencies": {
+    "@fluvio/client": "^5.0.0"
+  }
 }
 ```
 
+{{< idea >}}
```
Comment (nicholastmosher) [inline content/tutorials/hello-world-node.md:120]: I'm not sure I'm a fan of putting this in an idea bubble

File: content/tutorials/hello-world-node.md [hunk 4]
```diff
@@ -35,320 +36,459 @@ Node.js installation varies depending on your operating system.
...
-  try {
-    const flvConnection = await FluvioClient.connect();
-    let replica = await flvConnection.replica("my-topic", 0);
-    let len = await replica.produce("test");
-    
-    console.log("OK: %d bytes sent", len);
-  } catch (e) {
-    console.log("error: ", e.msg());
-  }
-}
```
Comment (nicholastmosher) [inline content/tutorials/hello-world-node.md:158]: Remove blank lines

**Comment by nicholastmosher** [comment]
This page seems too heavyweight for a "Hello World" app. How can we minimize the setup that a user needs to have? Do they need to have a full-fledged npm package setup? Do they need to have a StreamingApp class? Are we able to minimize this to the point that it could work in a node REPL?

File: content/tutorials/hello-world-node.md [hunk 5]
```diff
@@ -35,320 +36,459 @@ Node.js installation varies depending on your operating system.
...
+  private async configure(
+    topicName: string,
+    partition?: number,
+    isProducer?: boolean
+  ) {
+    console.log("Connecting to Fluvio Cluster");
+    // Setup the connection for the fluvio client;
+    await this.fluvio.connect();
+
+    if (isProducer) {
```
Comment (sehz) [inline content/tutorials/hello-world-node.md:212]: I think this is too much for hello world.  It should not have unnecessary steps for checking.  Just create producer

Comment (sehz) [reply content/tutorials/hello-world-node.md:212]: No admin API should be used in the hello world example

File: content/tutorials/hello-world-node.md [hunk 6]
```diff
@@ -35,320 +36,459 @@ Node.js installation varies depending on your operating system.
...
+
+// Create a StreamingApp Class to encapsulate our logic;
+export default class StreamingApp {
+  // The StreamingApp class has a `fluvio` client property
+  // to produce and consumer topics;
+  fluvio: Fluvio;
+
+  // The StreamingApp class has an `admin` FluvioAdmin
+  // to manage topics;
+  admin?: FluvioAdmin;
```
Comment (sehz) [inline content/tutorials/hello-world-node.md:321]: remove admin

**Review by sehz — CHANGES_REQUESTED** [review]
Let's try to simplify much as possible.  
Hello world should have very minimum of code.  
Can introduce more functionality in the next tutorial rater than trying to do here.

File: content/tutorials/hello-world-node.md (hunk 5 above)
Comment (nicholastmosher) [reply content/tutorials/hello-world-node.md:212]: Yeah, I think it's perfectly fine for the Hello World example to lean on the CLI for admin functionality. You can have them run commands like `fluvio topic create greetings` as part of the setup with no trouble. That will simplify the code they see.

File: content/tutorials/hello-world-node.md (hunk 6 above)
Comment (Ryanmtate) [reply content/tutorials/hello-world-node.md]: commit 3a13337 replaces with condensed example;

File: content/tutorials/hello-world-node.md (hunk 5 above)
Comment (Ryanmtate) [reply content/tutorials/hello-world-node.md]: commit 3a13337 uses fluvio cli to create topic

**Comment by Ryanmtate** [comment]
> This page seems too heavyweight for a "Hello World" app. How can we minimize the setup that a user needs to have? Do they need to have a full-fledged npm package setup? Do they need to have a StreamingApp class? Are we able to minimize this to the point that it could work in a node REPL?

commit 3a13337 replaces with condensed example;

File: content/tutorials/hello-world-node.md (hunk 1 above)
Comment (Ryanmtate) [reply content/tutorials/hello-world-node.md]: commit 3a13337 now includes comments in plain text

File: content/tutorials/hello-world-node.md (hunk 2 above)
Comment (Ryanmtate) [reply content/tutorials/hello-world-node.md]: commit 3a13337 fixes;

File: content/tutorials/hello-world-node.md (hunk 3 above)
Comment (Ryanmtate) [reply content/tutorials/hello-world-node.md]: commit 3a13337 now uses `{{< idea >}}` to illustrate code comments;

File: content/tutorials/hello-world-node.md (hunk 4 above)
Comment (Ryanmtate) [reply content/tutorials/hello-world-node.md:65]: commit 91158f0 attempts to clean up blank lines; not sure if there is a formatter out there that we could use to automate this.

File: content/tutorials/hello-world-node.md [hunk 7]
```diff
@@ -1,28 +1,31 @@
...
-**Prerequisites:** Examples in this section require an existing Fluvio cluster and a topic named "my-topic".<br> Step-by-step instructions are available in [Quick Start](/docs/getting-started/) at:
+* [Create a local Fluvio cluster](/docs/getting-started/install-local/)
 
-* [Create a cluster on Fluvio Cloud](/docs/getting-started/#create-a-fluvio-cloud-account)
-* [Add a topic](/docs/getting-started/#create-a-topic-and-stream-hello-world)
+### Create a Topic using the Fluvio CLI
 
-{{< /idea >}}
+```bash
+fluvio topic create -c 127.0.0.1:9003 -p 1 -r 1 my-topic
```
Comment (sehz) [inline content/tutorials/hello-world-node.md:23]: remove `-C` option. not needed

File: content/tutorials/hello-world-node.md [hunk 8]
```diff
@@ -33,324 +36,183 @@ Node.js installation varies depending on your operating system.
...
 
-### Start a new Node project
+```TypeScript
+import Fluvio from "@fluvio/client";
+import { createInterface } from "readline";
 
-Create a directory for **fluvio-node-app**:
+// Create Fluvio Client Instance
+const fluvio = new Fluvio({
+  host: "127.0.0.1",
```
Comment (sehz) [inline content/tutorials/hello-world-node.md:91]: host/port is not needed

File: content/tutorials/hello-world-node.md [hunk 9]
```diff
@@ -33,324 +36,183 @@ Node.js installation varies depending on your operating system.
...
-A dependency to @fluvio/client is added to package.json. 
 
-### Build "Hello Word" Streaming App
+```TypeScript
+import Fluvio, { OffsetFrom } from "@fluvio/client";
 
-Fluvio client needs a [default profile](/docs/cli/profiles) to identify the location and the authorization token of the cluster. The file was generated during cluster setup and it is available for download in your <a href="https://app.fluvio.io" target="_blank">Fluvio Cloud</a> account.
+// Create Fluvio Client Instance
+const fluvio = new Fluvio({
+  host: "127.0.0.1",
```
Comment (sehz) [inline content/tutorials/hello-world-node.md:137]: host/port is not needed

**Review by sehz — CHANGES_REQUESTED** [review]

File: content/tutorials/hello-world-node.md (hunk 7 above)
Comment (Ryanmtate) [reply content/tutorials/hello-world-node.md]: commit 339932b fixes;

File: content/tutorials/hello-world-node.md (hunk 8 above)
Comment (Ryanmtate) [reply content/tutorials/hello-world-node.md]: commit 339932b fixes;

File: content/tutorials/hello-world-node.md (hunk 9 above)
Comment (Ryanmtate) [reply content/tutorials/hello-world-node.md]: commit 339932b fixes;

**Pull request merged**